### ISSUE TITLE
Crashed on copying and pasting a drawing.

### ISSUE THREAD

**Issue by Spongeyboi** [xournalpp/xournalpp#5206]
### Operating System

Linux

### (Linux only) Distribution

Arch

### (Linux only) Desktop Environment

Sway

### (Linux Only) Display Server

Wayland

### Installation Method

Arch extra repository

### Xournal++ Version

xournalpp 1.2.1 (b48932a2)

### libgtk Version

3.24.38

### Bug Description

I tried copying a drawing I made with a lot of lines in different colours and pasting it in a different page. The drawing pasted, but after appearing on the page, Xournalpp would crash.

### Expected Behaviour

That it would not crash.

### Steps to Reproduce

If the logs aren't enough, I can send a Mega nz link to the .xopp to see if others get the same error on it.

### Additional Context

[errorlog.20231006-171825.log](https://github.com/xournalpp/xournalpp/files/12833395/errorlog.20231006-171825.log)

**Comment by rolandlo** [comment]
I can reproduce the issue (open a long PDF, copy something from one page, scroll down quickly to another page, paste). It's a failing assertion:
```
(com.github.xournalpp.xournalpp:22686): xopp-CRITICAL **: 20:23:05.952: Assertion failed: view->buffer.isInitialized()
    in function rerenderRectangle
    at line 49 of ./src/core/control/jobs/RenderJob.cpp
```

It's the same underlying cause like #5046, I think.

**Comment by bhennion** [comment]
Fixed via #5144
